### PR TITLE
✏ implement getScanDates()

### PR DESCRIPTION
**Pull request by kakashi250802**
add a function return array date of the semester

### CODE REVIEW

File: static/js/screens/results.js [hunk 1]
```diff
@@ -6,258 +6,340 @@
...
+
+
+        /**
+         * 
+         * @param	{*}			param0 
+         * @returns {String[]}	array day of 2 week // format :2021-12-06
+         */
+
+        getScanDates(year, semester, count) {
+            var DateArr = []
```
Comment (Belikhun) [inline static/js/screens/results.js:250]: Avoid using `var`, please use `let` to declare a variable in current scope.

File: static/js/screens/results.js [hunk 2]
```diff
@@ -6,258 +6,340 @@
...
+                DateArr = [...DateArr, Databegin];
+            } else {
+                var Databegin = new Date(`July 7, ${year}`);
+                Databegin = new Date(Databegin.setDate(Databegin.getDate() - Databegin.getDay() + 1));
+                for (let i = 0; i < count; ++i) {
+                    DateArr = [...DateArr, Databegin];
+                    Databegin = new Date(Databegin.setDate(Databegin.getDate() + 7));
+                }
+            }
+            return DateArr
```
Comment (Belikhun) [inline static/js/screens/results.js:273]: Variable naming should follow Camel Case naming rule. For example, `DateArr` -> `dateArr`.
Please refer to [this](https://topdev.vn/blog/quy-chuan-dat-ten-trong-lap-trinh-camelcase-underscore-hay-pascalcase/).

**Review by Belikhun — CHANGES_REQUESTED** [review]

File: static/js/screens/results.js (hunk 1 above)
Comment (Belikhun) [reply static/js/screens/results.js:250]: Also change this variable name to `dates` for beter clarity.

File: static/js/screens/results.js [hunk 3]
```diff
@@ -6,258 +6,340 @@
...
+        /**
+         * 
+         * @param	{*}			param0 
+         * @returns {String[]}	array day of 2 week // format :2021-12-06
+         */
+
+        getScanDates(year, semester, count) {
+            var DateArr = []
+            if (semester === 1) {
+                var Databegin = new Date(`October 1, ${year}`);
```
Comment (Belikhun) [inline static/js/screens/results.js:252]: This variable should be named `beginDate`

**Review by Belikhun — CHANGES_REQUESTED** [review]

File: static/js/screens/results.js [hunk 4]
```diff
@@ -6,258 +6,340 @@
...
+        },
+
+
+
+        /**
+         * 
+         * @param	{*}			param0 
+         * @returns {String[]}	array day of 2 week // format :2021-12-06
+         */
+
```
Comment (Belikhun) [inline static/js/screens/results.js:248]: Remove this empty line to make the jsDoc belong to the function below.

File: static/js/screens/results.js [hunk 5]
```diff
@@ -6,258 +6,340 @@
...
+            }
+        },
+
+
+
+        /**
+         * 
+         * @param	{*}			param0 
+         * @returns {String[]}	array day of 2 week // format :2021-12-06
+         */
```
Comment (Belikhun) [inline static/js/screens/results.js:247]: Too basic jsDoc, add something 😑

**Review by Belikhun — CHANGES_REQUESTED** [review]

File: static/js/screens/results.js (hunk 1 above)
Comment (kakashi250802) [reply static/js/screens/results.js:250]: oke bro

File: static/js/screens/results.js (hunk 4 above)
Comment (kakashi250802) [reply static/js/screens/results.js:248]: hummm

**Comment by kakashi250802** [comment]
we will update all

File: static/js/screens/results.js [hunk 6]
```diff
@@ -247,30 +247,30 @@ core.screen = {
...
             } else if (semester === 2) {
-                var Databegin = new Date(`January 12, ${year}`);
-                Databegin = new Date(Databegin.setDate(Databegin.getDate() - Databegin.getDay() + 1));
-                DateArr = [...DateArr, Databegin];
-                Databegin = new Date(`March 3, ${year}`);
-                Databegin = new Date(Databegin.setDate(Databegin.getDate() - Databegin.getDay() + 1));
-                DateArr = [...DateArr, Databegin];
+                let dateDefault = new Date(`January 12, ${year}`);
+                dateDefault = new Date(dateDefault.setDate(dateDefault.getDate() - dateDefault.getDay() + 1));
+                dates = [...dates, dateDefault];
```
Comment (Belikhun) [inline static/js/screens/results.js:256]: I think by this you meant to use `dates.push(dateDefault)`. This is not very performance efficient.

Comment (kakashi250802) [reply static/js/screens/results.js:256]: oke

**Comment by kakashi250802** [comment]
js doc  updated in last update of me see it

File: static/js/screens/results.js (hunk 5 above)
Comment (kakashi250802) [reply static/js/screens/results.js]: image

**Pull request merged**